CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+
+You must clone the Nerosim-lab/netpyne repository from your remote fork, and also connect the local copy to the `upstream` version of the codebase, so you can stay up-to-date with changes from other contributors.
+
+Make a local clone of your remote fork (`origin`):
+
+```bash
+ ~$ cd $INSTALL_LOCATION
+ ~$ git clone https://github.com/$GITHUB_USERNAME/netpyne.git
+```
+
+> **Remote URLs in git**
+>Here we use `git://` instead of `https://` in the URL for the `upstream` remote repository. `git://` URLs are read-only, so you can `pull` changes from `upstream` into your local copy (to stay up-to-date with changes from other contributors) but you cannot `push` changes from your computer into the `upstream` remote. Instead, you must `push` your changes to your own remote fork (`origin`) first, and then create a pull request from your remote into the `upstream` remote. In [a later section](#connecting-to-github-with-ssh-optional) you’ll see a third kind of remote URL for connecting to GitHub using SSH.
+
+Finally, set up a link between your local clone and the official repository (`upstream`):
+
+```bash
+~$ cd netpyne
+~$ git remote add upstream https://github.com/Neurosim-lab/netpyne.git
+~$ git fetch --all
+~$ git remote -v   #Shows your tracked repositories.
+```
+
+### Basic git commands
+
+Learning to work with git can take a long time, because it is a complex and powerful tool for managing versions of files across multiple users, each of whom have multiple copies of the codebase. We’ve already seen in the setup commands above a few useful git commands:
+
+* `git clone <URL_OF_REMOTE_REPO>` (make a local copy of a repository).
+* `git remote add <NICKNAME_OF_REMOTE> <URL_OF_REMOTE_REPO>` (connect a local copy to an additional remote).
+* `git fetch --all` (get the current state of connected remote repos).
+
+Other commands that you will undoubtedly need related to branches can be found [here](https://help.github.com/en/articles/about-branches). Branches represent multiple copies of the codebase within a local clone or remote repo. Branches are typically used to experiment with new features while still keeping a clean, working copy of the original codebase that you can switch back to at any time. The default branch of any repo is always called `master`, and it is recommended that you reserve the `master` branch to be that clean copy of the working `upstream` codebase. 
+
+In Neurosim-lab/netpyne repo, you must not work in the `master` branch, there is a branch called `development` for contribuiting. Therefore, if you want to add a new change, you should first synchronize your local `development` branch with the `upstream` repository, then create a new branch based on `development` and [check it out](https://git-scm.com/docs/git-checkout) so that any changes you make will exist on that new branch:

```

Comment:
I guess a simpler option is to make changes on the 'development' branch of the fork and then do a PR directly from there to the upstream 'development' branch?

Reply:
Yes, but here explain how to create a new branch, then how to add the changes (git add) and commit them, and finally says how to push and do a PR. 
So, I propose you change two paragraphs:

**This**
In Neurosim-lab/netpyne repo, you must not work in the `master` branch, there is a branch called `development` for contribuiting. Therefore, if you want to add a new change, you should first synchronize your local `development` branch with the `upstream` repository, then make your changes in `development` branch. Also you can create a new branch based on `development` and [check it out](https://git-scm.com/docs/git-checkout) so that any changes you make will exist on that new branch:

```bash
 ~$ git checkout development            # switch to local development branch
 ~$ git fetch upstream                  # get the current state of the remote upstream repo
 ~$ git merge upstream/development      # synchronize local development branch with remote upstream development branch
 ~$ git checkout -b new-branch          # create local branch "new-branch" and check it out
```

> **Alternative**
> You can save some typing by using `~$ git pull upstream/development` to replace the `fetch` and `merge` lines above.

> **Note:** You could synchronize your local `master` branch with that of the `upstream` repository to have the updated version, but you **_must never_** make any changes on the `master` branch.

**And this**
Now, it’s time to organize your changes into a series of [commits](https://git-scm.com/docs/git-commit). Commits are like snapshots of the repository — actually, more like a description of what has to change to get from the most recent snapshot to the current snapshot.